### PR TITLE
Fix unit handling in continuum and moment calculations

### PR DESCRIPTION
**Pull request by rosteen**
Fixes calculating moment 0 when display unit is in Hz, and more generally uses display unit more consistently in all continuum calculations.

### CODE REVIEW

File: jdaviz/core/template_mixin.py
Comment (kecnry) [inline jdaviz/core/template_mixin.py:2974]: I feel like @cshanahan1 had just removed this.  Was there a reason for that or did it just not seem to affect that use-case (see https://github.com/spacetelescope/jdaviz/pull/3211/files#r1787615549)

**Comment by gibsongreen** [comment]
In testing I am still seeing the traceback occur for moment zero (and it is occurring if you chose any spectral unit we allow).

**Comment by rosteen** [comment]
> In testing I am still seeing the traceback occur for moment zero (and it is occurring if you chose any spectral unit we allow).

Can you give some more detail about your workflow? I see everything working fine on my end:

https://github.com/user-attachments/assets/a323574b-abc7-4ec8-be30-2b0378949b38

**Comment by rosteen** [comment]
Oh, that's not the same workflow as I used, you don't have a spectral region or continuum selected. I can't reproduce the error you're seeing though even without a region or continuum, although I do see an `IndexError` if I select a spectral region but have `None` for continuum in Hz. So I'll at least start looking into that.

**Comment by rosteen** [comment]
Turns out the error I was seeing is due to a bug in specutils, see https://github.com/astropy/specutils/pull/1187. We'll need to update the specutils pin after I release 1.18.0, I can do that here before we merge this.

**Review by gibsongreen — APPROVED** [review]
The tracebacks I was seeing came from my end, all resolved and working now! With the workflow you provided I tried many combinations with the example notebook and with a flux cube. The main notes I had I think are all outside of scope and are being addressed in follow ups and deal with the spectral density or pix2 handling.

**Review by cshanahan1 — APPROVED** [review]
I tested this with a cube in Jy / pix2 as well as Jy / sr and both work, including the continuum preview. My only take it or leave it suggestion would be to include changing spectral units and slice value in tests to make sure that those changes are reflected in the calculation.

**Comment by rosteen** [comment]
Thanks for the reviews, I just bumped the specutils pin to the new release. I'll merge once tests pass.

**Pull request merged**
